src/lib/storage.ts

```diff
@@ -0,0 +1,30 @@
+import type { Writable } from 'svelte/store';
+import { writable, get } from 'svelte/store';
+
+export type StoreName = `${'CHAT' | 'THEME'}_STORE`;
+
+export const createStorage = <T>(key: StoreName, initValue: T): Writable<T> => {
+    const storage = writable(initValue);
+
+    const storedValueStr = localStorage.getItem(key);
+    if (storedValueStr != null) storage.set(JSON.parse(storedValueStr));
+
+    storage.subscribe(($storage) => {
+        if ([null, undefined].includes($storage)) {
+            localStorage.removeItem(key);
+        } else {
+            localStorage.setItem(key, JSON.stringify($storage));
+        }
+    });
+
+    // TODO -- need to add a way to remove the listener
```

Comment:
`createStorage` could return an `unsubscribe` function that the caller can then use when appropriate.

```ts
const { storage, unsubscribe } = createStorage('CHAT_STORE', initVal)

// ...
onDestroy(unsubscribe)
```